gradebookng/tool/src/webapp/scripts/gradebook-grades.js

```diff
@@ -926,6 +926,8 @@ GradebookSpreadsheet.prototype._refreshColumnOrder = function() {
 
   self._COLUMN_ORDER = self.$table.find("thead tr th.gb-grade-item-column-cell").map(function() {
     return $(this).data("model");
+  }).sort(function(a, b) {
+    return a.getDefaultOrder() > b.getDefaultOrder();
   });
 
```

Comment:
COLUMN_ORDER in the JavaScript needs to be in the uncategorised order as it is used to reorder the columns as the "Enable Categories" button is clicked to disable categories.